lib/puppet_x/puppetlabs/dsc_type_helpers.rb

```diff
@@ -126,6 +126,24 @@ def self.validate_mof_type(mof_type, embeddedinstance_name, name, value)
           end
         end
       end
+
+      def self.reboot_relationship(resource)
+        reboot_resource = resource.catalog.resource(:reboot, 'dsc_reboot')
+        return [] if !reboot_resource
+
+        # if an edge already exists from Reboot[dsc_reboot] to this resource, we have problems
```

Comment:
So the current code allows the run to proceed by not connecting the edge that would create the cyclic dependency.

If we try to create the cyclic dependency, then the run dies with:

```
Error: Failed to apply catalog: Found 1 dependency cycle:
(Dsc_file[foo] => Reboot[dsc_reboot] => Dsc_file[foo])
```

So if we're going to recognize the cycle, we should definitely fail I think.  The question is, do we bother creating a better msg for the user, or just let the standard Puppet error bubble up to them?